app/src/main/kotlin/com/wire/android/ui/userprofile/image/AvatarPickerViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.wire.android.ui.userprofile.image
+
+import android.content.Context
+import android.net.Uri
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.datastore.UserDataStore
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.util.toByteArray
+import com.wire.kalium.logic.data.user.UserAssetId
+import com.wire.kalium.logic.feature.asset.GetPublicAssetUseCase
+import com.wire.kalium.logic.feature.asset.PublicAssetResult
+import com.wire.kalium.logic.feature.user.UploadUserAvatarUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@ExperimentalMaterial3Api
+@HiltViewModel
+class AvatarPickerViewModel @Inject constructor(
+    private val navigationManager: NavigationManager,
+    private val dataStore: UserDataStore,
+    private val getUserAvatar: GetPublicAssetUseCase,
+    private val uploadUserAvatar: UploadUserAvatarUseCase,
+) : ViewModel() {
+
+    var avatarRaw by mutableStateOf<ByteArray?>(null)
+        private set
+
+    init {
+        viewModelScope.launch {
+            val assetId = dataStore.avatarAssetId.first()
+            avatarRaw = assetId?.let { loadAvatar(it) }
+        }
+    }
+
+    private suspend fun loadAvatar(avatarAssetId: UserAssetId): ByteArray? =
+        try {
+            (getUserAvatar(avatarAssetId) as PublicAssetResult.Success).asset
+        } catch (e: Exception) {
+            null
+        }
+
+    fun uploadNewPickedAvatarAndBack(chosenImgUri: Uri, context: Context) {
+        viewModelScope.launch {
+            withContext(Dispatchers.IO) {
+                val data = chosenImgUri.toByteArray(context)
+                uploadUserAvatar(mimeType = "image/jpg", imageData = data)
```

Comment:
I think for now it's ok doing it fixed to "image/jpg", but later on (for other types of assets) we could manage to extract the actual mimetype from the Uri, maybe as an Extension Function on `FileUtil` class. 
Using: 
- https://developer.android.com/reference/android/webkit/MimeTypeMap#getFileExtensionFromUrl(java.lang.String)
- https://developer.android.com/reference/android/webkit/MimeTypeMap#getMimeTypeFromExtension(java.lang.String)